experimental/packages/opentelemetry-sdk-metrics-base/src/MeterProvider.ts

```diff
@@ -69,9 +129,41 @@ export class MeterProvider implements metrics.MeterProvider {
     this._sharedState.metricCollectors.push(collector);
   }
 
-  addView(view: View, instrumentSelector: InstrumentSelector, meterSelector: MeterSelector) {
-    // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#view
-    this._sharedState.viewRegistry.addView(view, instrumentSelector, meterSelector);
+  addView(options: ViewOptions, selectorOptions?: SelectorOptions) {
+    if (isViewOptionsEmpty(options)) {
+      throw new Error('Cannot create view with no view arguments supplied');
+    }
+
+    // the SDK MUST NOT allow Views with a specified name to be declared with instrument selectors that select by instrument type or wildcard
```

Comment:
Basically, this statement looks to me that when view.name presents, we should not allow selectors that could potentially select multiple instruments. One can only select multiple instruments only when `selectorOptions.instrument.name` is not present or it is a wildcard pattern. All other cases like selecting with meters or instrument type should not be allowed.

Probably we need to update the spec words to:

```
the SDK MUST NOT allow Views with a specified name to be declared with instrument selectors that select by instrument type or wildcard or with only meter selectors.
```

So here the assertion could be simplified as:

```js
if (options.name != null && 
  (selectorOptions.instrument?.name == null || 
    PatternPredicate.hasWildcard(selectorOptions.instrument.name))) {
```

Reply:
This is correct, thanks! I updated this in [80503a4](https://github.com/open-telemetry/opentelemetry-js/pull/2820/commits/80503a47a5dd0569434cce9af0499234a6b3cc3d).

The spec comment was an oversight on my part. It has been updated in the past few weeks and now reads:

> In order to avoid conflicts, views which specify a name SHOULD have an instrument selector that selects at most one instrument. For the registration mechanism described above, where selection is provided via configuration, the SDK SHOULD NOT allow Views with a specified name to be declared with instrument selectors that may select more than one instrument (e.g. wild card instrument name) in the same Meter.

I changed the comment to use the current wording. :slightly_smiling_face: